apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -98,13 +103,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 return;
             }
 
+            var jsonStruct = new Struct();
+            jsonStruct.Fields.Add("message", Value.ForString(message));
+            jsonStruct.Fields.Add("log_name", Value.ForString(_logName));
+
+            if (GoogleLoggerScope.Current != null)
+            {
+                jsonStruct.Fields.Add("scope", Value.ForString(GoogleLoggerScope.Current.ToString()));
+            }
+
+            if (eventId.Id != 0 || eventId.Name != null)
+            {
+                var eventStruct = new Struct();
+                if (eventId.Id != 0)
+                {
+                    eventStruct.Fields.Add("id", Value.ForNumber(eventId.Id));
+                }
+                if (!string.IsNullOrWhiteSpace(eventId.Name))
+                {
+                    eventStruct.Fields.Add("name", Value.ForString(eventId.Name));
+                }
+                jsonStruct.Fields.Add("event_id", Value.ForStruct(eventStruct));
+            }
+
+            // If we have format params and its more than just the original message add them.
+            if (state is IEnumerable<KeyValuePair<string, object>> formatParams && 
+                !(formatParams.Count() == 1 && formatParams.Single().Key.Equals("{OriginalFormat}")))
+            {
+                var paramStruct = new Struct();
+                foreach (var pair in formatParams)
+                {
+                    paramStruct.Fields.Add(pair.Key, Value.ForString(pair.Value?.ToString() ?? ""));
```

Comment:
I wonder whether we should detect whether the value is formattable, and ask it to format itself in the invariant culture if so?

Reply:
I don't fully understand.  Do you mean check if the string is the format string? 

The fully formatted string is added in the 'message' feild

Reply:
Yea I think this is overkill.  The formatting of the full message is all handled above I don't think we need to worry about this here.

Reply:
Okay. It's probably worth adding a comment (not quite a TODO) as a prompt in case we come back here. I can see how it could be confusing in some cases. (Classic case: if you're running in a culture of ar-SA and you log a date, it'll come out in an entirely different calendar system...)

Reply:
Done.